js/init.js

```diff
@@ -1,32 +1,48 @@
 function syncState() {
-  chrome.storage.sync.get('classList', function(result) {
+  let board_id = location.pathname.split('/')[2];
+  let storageKey = 'classList_' + board_id;
+
+  chrome.storage.sync.get(storageKey, function(result) {
     let board = document.getElementById('board');
 
-    if (result.classList) {
-      board.classList.add(result.classList);
+    if (result[storageKey]) {
+      board.classList.add(result[storageKey]);
     }
   });
 }
 
+chrome.storage.onChanged.addListener(function (changes, namespace) {
+  let board = document.getElementById('board');
+  let board_id = location.pathname.split('/')[2];
+  let storageKey = 'classList_' + board_id;
+
+  if (namespace == 'sync' && storageKey in changes) {
+    board.classList.remove(changes[storageKey].oldValue);
+    board.classList.add(changes[storageKey].newValue);
+  }
+});
+
 function toggleLayout() {
   const classVertical = 'layout-trello-vertical';
   const classMixed = 'layout-trello-mixed';
 
   let board = document.getElementById('board');
+  let board_id = location.pathname.split('/')[2];
+  let storageKey = 'classList_' + board_id;
 
   if (board.classList.contains(classMixed)) {
     board.classList.remove(classMixed);
     board.classList.add(classVertical);
     chrome.storage.sync.set({
```

Comment:
Does this line now trigger the `chrome.storage.onChanged` listener, causing the class names to be added and removed a second time? If so, we can probably remove some unnecessary code in `toggleLayout`, but that's not super important.

Reply:
You're right! Done.